File: leetcode/24. uniqueMorseReps/uniqueMorseReps.py
from typing import List

def uniqueMorseRepresentations(words: List[str]) -> int:
    morseList = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---",
    "-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."
    ]
    res = set()
    for word in words:
        morse = ""
        for char in word:
            morse += morseList[ord(char)-97]
        res.add(morse)
    return len(res)


def main():
    words = ["gin", "zen", "gig", "msg"]
    uniqueMorses = uniqueMorseRepresentations(words)
    print(uniqueMorses)


if __name__ == "__main__":
    main()

File: leetcode/24. uniqueMorseReps/uniqueMorseReps.go
package main

import "fmt"

func uniqueMorseRepresentations(words []string) int {
	morselist := []string{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}
	morseset := make(map[string]bool, 0)
	for _, word := range words {
		morse := ""
		for _, rune := range word {
			morse += morselist[int(rune) - 97]
		}
		morseset[morse] = true
	}
	return len(morseset)
}

func main() {
	words := []string{"gin", "zen", "gig", "msg"}
	uniquecount := uniqueMorseRepresentations(words)
	fmt.Println(uniquecount)
}
